Lib/test/test_mmap.py

```diff
@@ -0,0 +1,864 @@
+    def test_crasher_on_windows(self):
+        # Should not crash (Issue 1733986)
+        m = mmap.mmap(-1, 1000, tagname="foo")
+        try:
+            mmap.mmap(-1, 5000, tagname="foo")[:] # same tagname, but larger size
+        except:
+            pass
+        m.close()
+
+        # Should not crash (Issue 5385)
+        with open(TESTFN, "wb") as fp:
+            fp.write(b"x"*10)
+        f = open(TESTFN, "r+b")
+        m = mmap.mmap(f.fileno(), 0)
+        f.close()
+        try:
+            m.resize(0) # will raise OSError
+        except:
+            pass
+        try:
+            m[:]
+        except:
+            pass
+        m.close()
+
+    @unittest.skipUnless(os.name == 'nt', 'requires Windows')
+    def test_invalid_descriptor(self):
+        # socket file descriptors are valid, but out of range
+        # for _get_osfhandle, causing a crash when validating the
+        # parameters to _get_osfhandle.
+        s = socket.socket()
+        try:
+            with self.assertRaises(OSError):
+                m = mmap.mmap(s.fileno(), 10)
+        finally:
+            s.close()
+
+    def test_context_manager(self):
+        with mmap.mmap(-1, 10) as m:
+            self.assertFalse(m.closed)
+        self.assertTrue(m.closed)
+
+    def test_context_manager_exception(self):
+        # Test that the OSError gets passed through
+        with self.assertRaises(Exception) as exc:
+            with mmap.mmap(-1, 10) as m:
+                raise OSError
+        self.assertIsInstance(exc.exception, OSError,
+                              "wrong exception raised in context manager")
+        self.assertTrue(m.closed, "context manager failed")
+
+    def test_weakref(self):
+        # Check mmap objects are weakrefable
+        mm = mmap.mmap(-1, 16)
+        wr = weakref.ref(mm)
+        self.assertIs(wr(), mm)
+        del mm
+        gc_collect()
+        self.assertIs(wr(), None)
+
+    def test_write_returning_the_number_of_bytes_written(self):
+        mm = mmap.mmap(-1, 16)
+        self.assertEqual(mm.write(b""), 0)
+        self.assertEqual(mm.write(b"x"), 1)
+        self.assertEqual(mm.write(b"yz"), 2)
+        self.assertEqual(mm.write(b"python"), 6)
+
+    @unittest.skipIf(os.name == 'nt', 'cannot resize anonymous mmaps on Windows')
+    def test_resize_past_pos(self):
+        m = mmap.mmap(-1, 8192)
+        self.addCleanup(m.close)
+        m.read(5000)
+        try:
+            m.resize(4096)
+        except SystemError:
+            self.skipTest("resizing not supported")
+        self.assertEqual(m.read(14), b'')
+        self.assertRaises(ValueError, m.read_byte)
+        self.assertRaises(ValueError, m.write_byte, 42)
+        self.assertRaises(ValueError, m.write, b'abc')
+
+    def test_concat_repeat_exception(self):
+        m = mmap.mmap(-1, 16)
+        with self.assertRaises(TypeError):
+            m + m
+        with self.assertRaises(TypeError):
+            m * 2
+
```

Comment:
I don't want to skip the whole test `test_flush_return_value`. I think `TODO: RUSTPYTHON` is enough.

Reply:
In this case, skipping the whole test is better. Our priority is `unittest.expectedFailure`, `unittest.skip` and comment the lowest.
If we use `unittest.expectedFailure`, the fix will be automatically caught. If we use `unittest.skip`, we will see there are skipped tests by looking at the result of test. About comments, we will never know it is commented out until directly looking at the code. 
Comment is only accepted for few cases, mostly when running code causes crash instead of test failure.

Reply:
What about let it fail?

Reply:
Yes, that's the best. If the test always fail, using `unittest.expectedFailure` will check it actually fail.

Reply:
Got it. But i think we `skipIf ` is better, because it accepts `condition` such as `sys.platform == "linux"`. This test will only fail on linux right now.

Reply:
Fixed in https://github.com/RustPython/RustPython/pull/3755/commits/9707df1ad1ed0af1cabe6a1d241b4225c51b2f77

Reply:
Make the condtion better https://github.com/RustPython/RustPython/pull/3755/commits/b98bfe2105c2c8f26c4ecc496533fe5495de9fdd

I think it's ok right now, please take a look.

Reply:
Yes, that's looking perfect!